src/components/Button/Button.tsx

```diff
@@ -1,12 +1,74 @@
 import React from "react";
 
-// Props
-type Props = {} & JSX.IntrinsicElements["button"];
+type IntrinsicElements = JSX.IntrinsicElements["button"];
 
-export const Button: React.FC<Props> = ({ children, ...rest }) => {
+interface Props extends IntrinsicElements {
+  buttonText: string;
+  buttonVariation?: ButtonVariation;
+  disabled?: boolean;
+  icon?: boolean;
+  shiftIconLeft?: boolean;
+  ariaLabel?: string;
+  largeButton?: boolean;
+}
+
+/**
+ * Button Component
+ * @param {string}  [buttonText]        Renders the text contained in the button.
+ * @param {string}  buttonVariation     Renders the style of the button.
+ * @param {string}  disabled            Determines whether or not a button is enabled.
+ * @param {string}  icon                Handles the icon to render in the button.
```

Comment:
Is this always going to be a string or should this accept an actual image/svg?

Reply:
Its a stringName now to use the Icon component